plugins/auto-sizes/hooks.php

```diff
@@ -139,79 +139,139 @@ function auto_sizes_get_width( string $layout_width, int $image_width ): string
 }
 
 /**
- * Filter the sizes attribute for images to improve the default calculation.
+ * Primes attachment into the cache with a single database query.
  *
- * @since 1.1.0
+ * @since n.e.x.t
  *
- * @param string                                                   $content      The block content about to be rendered.
- * @param array{ attrs?: array{ align?: string, width?: string } } $parsed_block The parsed block.
- * @return string The updated block content.
+ * @param string $content The HTML content.
+ * @param string $context Optional. Additional context to pass to the filters.
+ *                        Defaults to `current_filter()` when not set.
+ * @return string The HTML content.
  */
-function auto_sizes_filter_image_tag( string $content, array $parsed_block ): string {
+function auto_sizes_prime_attachment_caches( string $content, string $context = null ): string {
+	if ( null === $context ) {
```

Comment:
To guard against an error in case someone filters `the_content` to return a non-string value.

```suggestion
function auto_sizes_prime_attachment_caches( $content, string $context = null ): string {
	if ( ! is_string( $content ) ) {
		$content = '';
	}
	if ( null === $context ) {
```